pkg/pool-stable-phantom/contracts/StablePhantomPool.sol

```diff
@@ -1034,10 +1046,21 @@ contract StablePhantomPool is IRateProvider, BaseGeneralPool, ProtocolFeeCache {
             super._isOwnerOnlyAction(actionId);
     }
 
+    // Convert from an index into an array including BPT (the Vault's registered token list), to an index
+    // into an array excluding BPT (usually from user input, such as amountsIn/Out).
+    // `index` should not be the BPT token index itself, or it would duplicate the preceeding index value.
     function _skipBptIndex(uint256 index) internal view returns (uint256) {
         return index < _bptIndex ? index : index.sub(1);
     }
 
+    // Convert from an index into an array excluding BPT (usually from user input, such as amountsIn/Out),
+    // to an index into an array excluding BPT (the Vault's registered token list).
+    // `index` should not be the BPT token index itself, if it is the last element, or this would return an
+    // invalid index.
```

Comment:
```suggestion
    // `index` must not be the BPT token index itself.
```
We could add a relatively cheap check for this.

Reply:
Added, along with tests. At least or the new "add" one, there is a case where we're passing it in from user input, so it could be sent in wrong.